### PR TITLE
added clearSelect and clearWhere to query builder

### PR DESCRIPTION
**Pull request by alechirsch**
Solution to #1779, also related to #854

### CODE REVIEW

**Comment by elhigu** [comment]
I hadn't checked this pull request when I made the comment about API naming in the other issue :D nice choice 👍

I'll let someone else to decide if this should be getting in. Code looks good, but I would like to see also tests for different nested subquery builders.

I agree that these 2 are the most common uses for clear variants, but there are much more parts in query builder that people might like to clear, like order by is one of my favorites, since in count queries for it can cause huge performance hit.

File: src/query/builder.js [hunk 1]
```diff
@@ -960,8 +972,14 @@ assign(Builder.prototype, {
       aggregateDistinct: aggregateDistinct || false
     });
     return this;
-  }
+  },
 
+  // Helper function for clearing or reseting a grouping type from the builder
+  _clearGrouping(grouping){
+    this._statements = reject(this._statements, function(statement){
```
Comment (elhigu) [inline src/query/builder.js]: Arrow functions are preferred in library code if using `function ()` is not necessary.

Comment (alechirsch) [inline src/query/builder.js]: Loud and clear,  I also prefer store notation, however I saw function syntax in many places in the apiece, was not sure which was preferred.

**Comment by alechirsch** [comment]
@elhigu I agree, we should have the capability to clear any grouping type. I wrote the helper function to support future clearing of different groupings. In fact I did implement it for all of the grouping types that I found in the api, but removed them for now since I did not have the time to write tests for all of them. 

I also was not sure if I caught all of the types, is there a documented list of the types?

File: src/query/builder.js (hunk 1 above)
Comment (alechirsch) [inline src/query/builder.js]: Arrow notation*, in the api*

**Comment by alechirsch** [comment]
@elhigu I have added some tests for subqueries

**Comment by alechirsch** [comment]
Closing with preference for #1912

**Pull request closed without merging**